### PR TITLE
[Testing, Tooling] refactor: in-memory network usage

### PR DESCRIPTION
**Pull request by bryanchriswhite**
### Human Summary

Refactors usage of in-memory network in tests to utilize updated in-memory network testutils.

## Issue

- #141 
- #263
- #264
- #265 
- #289
- #290 
- :point_right: THIS PR :point_left: 
- #266 
- #250

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Run all unit tests**: `make go_develop_and_test`
- [ ] **Run E2E tests locally**: `make test_e2e`
- [ ] **Run E2E tests on DevNet**: Add the `devnet-test-e2e` label to the PR. This is VERY expensive, only do it after all the reviews are complete.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [x] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: x/application/client/cli/tx_delegate_to_gateway_test.go [hunk 1]
```diff
@@ -48,39 +51,35 @@ func TestCLI_DelegateToGateway(t *testing.T) {
 	}{
 		{
 			desc:           "delegate to gateway: valid",
-			appAddress:     appAccount.Address.String(),
-			gatewayAddress: gatewayAccount.Address.String(),
+			appAddress:     appBech32,
```
Comment (Olshansk) [inline x/application/client/cli/tx_delegate_to_gateway_test.go:54]: I think we should one of `appAddr` or `appAddrBech32` to make the variable name more semantically clear.

File: x/gateway/client/cli/tx_stake_gateway_test.go [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
-	"github.com/pokt-network/poktroll/x/gateway/types"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
 )
 
 func TestCLI_StakeGateway(t *testing.T) {
-	net, _ := networkWithGatewayObjects(t, 2)
-	val := net.Validators[0]
-	ctx := val.ClientCtx
+	ctx := context.Background()
+	memnet := gatewaynet.NewInMemoryNetworkWithGateways(
```
Comment (Olshansk) [inline x/gateway/client/cli/tx_stake_gateway_test.go:24]: Really like seeing this pattern everywhere.

**Review by Olshansk — CHANGES_REQUESTED** [review]
A few small comments. 

Let's merge with main, resolve/ignore the comments, run e2e tests and get this in!

File: testutil/network/basenet/accounts.go [hunk 3]
```diff
@@ -194,3 +194,22 @@ func (memnet *BaseInMemoryNetwork) FundGatewayAccounts(
...
+// in-memory network keyring). It initializes the address on-chain by sending
+// it some tokens and also creates a corresponding keypair in the in-memory
+// network's keyring. It returns the pre-generated account which was used.
+func (memnet *BaseInMemoryNetwork) CreateNewOnChainAccount(t *testing.T) *testkeyring.PreGeneratedAccount {
+	t.Helper()
+
+	preGeneratedAcct, ok := testkeyring.PreGeneratedAccounts().Next()
+	require.Truef(t, ok, "no pre-generated accounts available")
+
+	// Create a supplier on-chain account.
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply testutil/network/basenet/accounts.go:207]: Will #PUC, this DOES NOT have anything to do with getting a public key on-chain. That only happens when a tx signed by this new account is received by the network (as commented elsewhere).

File: testutil/testclient/localnet.go [hunk 4]
```diff
@@ -28,8 +27,6 @@ var (
 
 // init initializes the SDK configuration upon package import.
 func init() {
-	cmd.InitSDKConfig()
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply testutil/testclient/localnet.go:31]: It was causing issues:
image

We need to be careful when where and how (i.e. `init` functions or not) we call this function as (AFAIK) there is currently no way to check whether it has already been called.

**Comment by bryanchriswhite** [comment]
Closing, superseded by #412 & #417.

**Pull request closed without merging**